extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1186,12 +1482,22 @@ private void checkTaskDuration() throws InterruptedException, ExecutionException
               return null;
             }
 
-            log.info("Setting endOffsets for tasks in taskGroup [%d] to %s and resuming", groupId, endOffsets);
-            for (final String taskId : setEndOffsetTaskIds) {
-              setEndOffsetFutures.add(taskClient.setEndOffsetsAsync(taskId, endOffsets, true));
-            }
-
             try {
+
+              if (endOffsets.equals(taskGroup.partitionOffsets)) {
```

Comment:
If this condition is true, it just logs a warning and then continues on with the same logic. Is this missing something?

Reply:
Good catch, however actually this condition should never be true, don't remember why I added this probably can be removed, currently just changed to return `endOffsets`